### PR TITLE
Chore: news

### PR DESCRIPTION
**Pull request by amordzon**

### CODE REVIEW

File: src/Controllers/news.controller.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+} from 'express';
+import axios from 'axios';
+import cheerio from 'cheerio';
+
+export const getNewsMFI: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+) => {
+    const HTMLData = await axios
```
Comment (jzapiorkowski) [inline src/Controllers/news.controller.ts:16]: put request in try catch block

File: src/Controllers/news.controller.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        .get('https://mfi.ug.edu.pl/wydzial/aktualnosci')
+        .then((res) => res.data)
```
Comment (jzapiorkowski) [inline src/Controllers/news.controller.ts:18]: don't mix async await with Promise.then(), stick with async await

File: src/Controllers/news.controller.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        .catch((err) => {
+            res.send(err);
+        });
+    const newsArray: {}[] = [];
```
Comment (jzapiorkowski) [inline src/Controllers/news.controller.ts:22]: it would be nice to create interface for news

File: src/Controllers/news.controller.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    const $ = cheerio.load(HTMLData);
+    const selectedElem: string = '.item-list > ul > li';
+    $(selectedElem).each((parentIndex, parentElem) => {
```
Comment (jzapiorkowski) [inline src/Controllers/news.controller.ts:25]: maybe theres a way to use reduce method instead of loop?

File: src/Controllers/news.controller.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    });
+    res.send(newsArray);
+};
+
+export const getNewsINF: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+) => {
+    const HTMLData = await axios
```
Comment (jzapiorkowski) [inline src/Controllers/news.controller.ts:50]: same as above

File: src/Controllers/news.controller.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        .get('https://inf.ug.edu.pl/studinfo', {
+            headers: {
+                'Accept-Encoding': 'application/json',
+            },
+        })
+        .then((res) => res.data)
```
Comment (jzapiorkowski) [inline src/Controllers/news.controller.ts:56]: same as above

File: src/Controllers/news.controller.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        .catch((err) => {
+            res.send(err);
+        });
+    const newsArray: {}[] = [];
```
Comment (jzapiorkowski) [inline src/Controllers/news.controller.ts:60]: same as above

File: src/Controllers/news.controller.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    const $ = cheerio.load(HTMLData);
+    const selectedElem: string = 'div.newsBox';
+    $(selectedElem).each((parentIndex, parentElem) => {
```
Comment (jzapiorkowski) [inline src/Controllers/news.controller.ts:63]: same as above

**Review by jzapiorkowski — COMMENTED** [review]
good work, but there are still things to work on tho

File: src/Controllers/news.controller.ts (hunk 4 above)
Comment (amordzon) [reply src/Controllers/news.controller.ts:25]: idk i searched the internet and it seems that everyone uses each or map with cheerio

**Pull request closed without merging**